src/nbl/asset/interchange/IAssetLoader.cpp

```diff
@@ -35,22 +35,36 @@ void IAssetLoader::IAssetLoaderOverride::insertAssetIntoCache(SAssetBundle& asse
         m_manager->insertAssetIntoCache(asset, ASSET_MUTABILITY_ON_CACHE_INSERT);
 }
 
-bool IAssetLoader::IAssetLoaderOverride::handleRestore(core::smart_refctd_ptr<IAsset>&& _chosenAsset, SAssetBundle& _bundle, SAssetBundle& _reloadedBundle, uint32_t _restoreLevels)
+void IAssetLoader::IAssetLoaderOverride::handleRestore(core::smart_refctd_ptr<IAsset>&& _chosenAsset, SAssetBundle& _bundle, SAssetBundle& _reloadedBundle, uint32_t _restoreLevels)
 {
     if (_bundle.getContents().size() != _reloadedBundle.getContents().size())
-        return false;
+        return;
 
-    auto dummies = _bundle.getMutableContents();
+    auto dummies = _bundle.getContents();
     auto found_it = std::find(dummies.begin(), dummies.end(), _chosenAsset);
     if (found_it == dummies.end())
-        return false;
+        return;
     const uint32_t ix = found_it - dummies.begin();
 
-    auto reloaded = _reloadedBundle.getMutableContents();
+    auto reloaded = _reloadedBundle.getContents();
     if (dummies.begin()[ix]->isADummyObjectForCache() && !dummies.begin()[ix]->canBeRestoredFrom(reloaded.begin()[ix].get()))
```

Comment:
little note, in this case dummies.begin()[ix] should literally be the same thing as _chosenAsset so just use that.